rln/src/pm_tree_adapter.rs

```diff
@@ -60,6 +60,89 @@ fn get_tmp() -> bool {
 
 pub struct PmtreeConfig(Config);
 
+pub struct PmtreeConfigBuilder {
```

Comment:
Does it make sense to make all the values optional? I would assume that either you provide all the values OR you deal with default values.